src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -628,47 +631,71 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
-                var request = new HttpRequestMessage(HttpMethod.Get, SlowStreamingServer);
-                using (HttpResponseMessage response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead))
+                Task<HttpResponseMessage> getResponse = client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+
+                await AcceptSocketAsync(server, async (s, stream, reader, writer) =>
                 {
-                    Stream stream = await response.Content.ReadAsStreamAsync();
-                    bytesRead = await stream.ReadAsync(buffer, 0, buffer.Length);
-                }
-                _output.WriteLine("Bytes read from stream: {0}", bytesRead);
-                Assert.True(bytesRead < buffer.Length, "bytesRead should be less than buffer.Length");
-            }            
+                    while (!string.IsNullOrEmpty(reader.ReadLine())) ;
+                    await writer.WriteAsync("HTTP/1.1 200 OK\r\nContent-Length: 16000\r\n\r\nless than 16000 bytes");
+                    await writer.FlushAsync();
+
+                    using (HttpResponseMessage response = await getResponse)
+                    {
+                        var buffer = new byte[8000];
+                        using (Stream clientStream = await response.Content.ReadAsStreamAsync())
+                        {
+                            int bytesRead = await clientStream.ReadAsync(buffer, 0, buffer.Length);
+                            _output.WriteLine($"Bytes read from stream: {bytesRead}");
+                            Assert.True(bytesRead < buffer.Length, "bytesRead should be less than buffer.Length");
+                        }
+                    }
+                });
+            });
         }
 
         [Fact]
         public async Task Dispose_DisposingHandlerCancelsActiveOperations()
         {
-            using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            await CreateServerAsync(async (handler, client, socket, url) =>
             {
-                socket.Bind(new IPEndPoint(IPAddress.Loopback, 0));
-                socket.Listen(1);
-                Task<Socket> accept = socket.AcceptAsync();
-
-                IPEndPoint local = (IPEndPoint)socket.LocalEndPoint;
-                var client = new HttpClient();
-                Task<string> download = client.GetStringAsync($"http://{local.Address}:{local.Port}");
-
-                using (Socket acceptedSocket = await accept)
+                Task<string> download = client.GetStringAsync(url);
+                using (Socket acceptedSocket = await socket.AcceptAsync())
                 {
                     client.Dispose();
                     await Assert.ThrowsAnyAsync<OperationCanceledException>(() => download);
                 }
+            });
+        }
+
+        private static async Task CreateServerAsync(Func<HttpClientHandler, HttpClient, Socket, Uri, Task> funcAsync)
```

Comment:
Can you put this loopback server into a separate file/static class?  It will help as we refactor this large HttpClientHandlerTest.cs file.  

There is already a LoopbackServer.cs file.  Can't this be blended into that file/class in a more generalized way?